src/app/app-routing.module.ts

```diff
@@ -1,10 +1,22 @@
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
-const routes: Routes = [];
+import { AppComponent } from './app.component';
+import { HeaderComponent } from './components/header/header.component';
+import { HomepageComponent } from './components/homepage/homepage.component';
+import { ListingpageComponent } from './components/listingpage/listingpage.component';
+import { ProductpageComponent } from './components/productpage/productpage.component';
+
+
+const routes: Routes = [
+  {path : "", component : HomepageComponent},
+  {path : "Home", component : HomepageComponent},
+  {path : "List", component : ListingpageComponent},
+  {path : "Product/:id", component : ProductpageComponent},
+];
 
 @NgModule({
   imports: [RouterModule.forRoot(routes)],
   exports: [RouterModule]
 })
-export class AppRoutingModule { }
+export class AppRoutingModule { }
```

Comment:
each file needs to have a line at the end of file - otherwise you see the red icon at the last line of the file - indicating this is missing. Ensure you use a formatter plugin - like prettier or eslint or editorconfig to handle this